jumanji/environments/routing/connector/generation_methods.py

```diff
@@ -0,0 +1,437 @@
+
+        # Create a correction mask for possible collisions (see the docs of `get_correction_mask`)
+        correction_fn = jax.vmap(get_correction_mask, in_axes=(None, None, 0))
+        correction_masks, collided_agents = correction_fn(grid, joined_grid, agent_ids)
+        correction_mask = jnp.sum(correction_masks, 0)
+
+        # Correct state.agents
+        # Get the correct agents, either old agents (if collision) or new agents if no collision
+        agents = jax.vmap(
+            lambda collided, old_agent, new_agent: jax.lax.cond(
+                collided,
+                lambda: old_agent,
+                lambda: new_agent,
+            )
+        )(collided_agents, agents, new_agents)
+        # Create the new grid by fixing old one with correction mask and adding the obstacles
+        return agents, joined_grid + correction_mask
+
+    def _initialise_agents(
+        self, key: chex.PRNGKey, grid: chex.Array
+    ) -> Tuple[chex.Array, Agent]:
+        """Initialises agents using random starting point and places heads on grid.
+
+        Args:
+            key: random key.
+            grid: empty grid.
+
+        Returns:
+            Tuple of grid with populated starting points and agents initialised with
+            the same starting points.
+        """
+        starts_flat = jax.random.choice(
+            key=key,
+            a=jnp.arange(self.rows * self.cols),
+            shape=(1, self.num_agents),
+            # Start positions for all agents
+            replace=False,
+        )
+
+        # Create 2D points from the flat arrays.
+        starts = jnp.divmod(starts_flat[0], self.cols)
+        # Fill target with default value as targets will be assigned aftert random walk
+        targets = jnp.full((2, self.num_agents), -1)
+
+        # Initialise agents
+        agents = jax.vmap(Agent)(
+            id=jnp.arange(self.num_agents),
+            start=jnp.stack(starts, axis=1),
+            target=jnp.stack(targets, axis=1),
+            position=jnp.stack(starts, axis=1),
+        )
+        # Place agent heads on grid
+        grid = jax.vmap(self._place_agent_heads_on_grid, in_axes=(None, 0))(
+            grid, agents
+        )
+        grid = grid.max(axis=0)
+        grid = jnp.array(grid, dtype=int)
+        return grid, agents
+
+    def _place_agent_heads_on_grid(self, grid: chex.Array, agent: Agent) -> chex.Array:
+        """Updates grid with agent starting positions."""
+        return grid.at[agent.start[0], agent.start[1]].set(get_position(agent.id))
+
+    def _continue_stepping(
+        self, stepping_tuple: Tuple[chex.PRNGKey, chex.Array, Agent]
+    ) -> chex.Array:
+        """Determines if agents can continue taking steps."""
+        key, grid, agents = stepping_tuple
+        dones = jax.vmap(self._is_any_step_possible, in_axes=(None, 0))(grid, agents)
+        return ~dones.all()
+
+    def _is_any_step_possible(self, grid: chex.Array, agent: Agent) -> chex.Array:
```

Comment:
To me, it seems this function is checking if a cell is stuck, as in is done. I would suggest renaming the method to `_no_available_cells`